tests/property/ic_api/cycles/src/cycles/index.ts

```diff
@@ -0,0 +1,65 @@
+import {
+    canisterBalance,
+    IDL,
+    msgCyclesAccept,
+    msgCyclesAvailable,
+    reply,
+    update
+} from 'azle';
+
+import { CyclesResult, CyclesResultIDL } from '../types';
+
+export default class {
+    @update([], CyclesResultIDL)
+    receiveAllCycles(): CyclesResult {
+        return acceptCycles((available) => available);
+    }
+
+    @update([], CyclesResultIDL)
+    receiveHalfCycles(): CyclesResult {
+        return acceptCycles((available) => available / 2n);
+    }
+
+    @update([], CyclesResultIDL)
+    receiveNoCycles(): CyclesResult {
+        return acceptCycles(() => 0n);
+    }
+
+    // TODO we could have fun playing with the constraints of msgCyclesAccept
```

Comment:
Consider how making the example more arbitrary like we have discussed would solve these comments

Reply:
done